oauth2client/clientsecrets.py

```diff
@@ -60,32 +60,39 @@
 
 class Error(Exception):
     """Base error for this module."""
-    pass
 
 
 class InvalidClientSecretsError(Error):
     """Format of ClientSecrets file is invalid."""
-    pass
 
 
-def _validate_clientsecrets(obj):
+def _validate_clientsecrets(clientsecrets_dict):
+    """Validate parsed client secrets from a file.
+
+    Args:
+        clientsecrets_dict: dict, a dictionary holding the client secrets.
+
+    Returns:
+        tuple, a string of the client type and the information parsed
+        from the file.
+    """
     _INVALID_FILE_FORMAT_MSG = (
         'Invalid file format. See '
         'https://developers.google.com/api-client-library/'
         'python/guide/aaa_client_secrets')
 
-    if obj is None:
+    if clientsecrets_dict is None:
         raise InvalidClientSecretsError(_INVALID_FILE_FORMAT_MSG)
-    if len(obj) != 1:
+    if len(clientsecrets_dict) != 1:
         raise InvalidClientSecretsError(
             _INVALID_FILE_FORMAT_MSG + ' '
             'Expected a JSON object with a single property for a "web" or '
             '"installed" application')
-    client_type = tuple(obj)[0]
+    client_type, = clientsecrets_dict.keys()
     if client_type not in VALID_CLIENT:
         raise InvalidClientSecretsError(
             'Unknown client type: %s.' % (client_type,))
-    client_info = obj[client_type]
```

Comment:
Was going for test coverage here rather than changing the existing (not so stellar) code. Happy to change if desired.

We actually check both and could just watch for a `ValueError`:

```python
>>> (a, b), = {1: 2, 3: 4}.items()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack
>>> (a, b), = {1: 2}.items()
>>> (a, b), = {}.items()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: need more than 0 values to unpack
```